docs/en/reference/events.rst

```diff
@@ -334,3 +334,85 @@ for event listeners.
 
 It allows you to access the table index definitions of the current database, table name, Platform and
 ``Doctrine\DBAL\Connection`` instance. Indexes, that are about to be added, are not listed.
+
+OnTransactionBegin Event
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Doctrine\DBAL\Events::onTransactionBegin`` is triggered on `Connection::beginTransaction()` is called.
+An instance of ``Doctrine\DBAL\Event\TransactionBeginEventArgs`` is injected as argument
+for event listeners.
+
+.. code-block:: php
+
+    <?php
+    class MyEventListener
+    {
+        public function onTransactionBegin(TransactionBeginEventArgs $event)
+        {
+            // Your EventListener code
+        }
+    }
+
+    $evm = new EventManager();
+    $evm->addEventListener(Events::onTransactionBegin, new MyEventListener());
+
+    $conn = DriverManager::getConnection($connectionParams, null, $evm);
+
+It allows you to access the``Doctrine\DBAL\Connection`` instance.
+Please note that this event can be called multiple times, since transactions can be nested.
+
+OnTransactionCommit Event
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Doctrine\DBAL\Events::onTransactionCommit`` is triggered on `Connection::commit()` is called.
+An instance of ``Doctrine\DBAL\Event\TransactionCommitEventArgs`` is injected as argument
+for event listeners.
+
+.. code-block:: php
+
+    <?php
+    class MyEventListener
+    {
+        public function onTransactionCommit(TransactionCommitEventArgs $event)
+        {
+            // Your EventListener code
+        }
+    }
+
+    $evm = new EventManager();
+    $evm->addEventListener(Events::onTransactionCommit, new MyEventListener());
+
+    $conn = DriverManager::getConnection($connectionParams, null, $evm);
+
+It allows you to access the``Doctrine\DBAL\Connection`` instance.
+Please note that this event can be called multiple times, since transactions can be nested.
+If you want to know if transaction is actually committed, you should rely on
+`TransactionCommitEventArgs->getConnection()->getTransactionNestingLevel() === 0`
```

Comment:
It probably should be `TransactionCommitEventArgs::` instead of `->` since it's a class name.